network/p2p/peerstore/peerstore.go

```diff
@@ -18,15 +18,41 @@ package peerstore
 
 import (
 	"fmt"
+	"math/rand"
+	"time"
 
 	"github.com/libp2p/go-libp2p/core/peer"
 	libp2p "github.com/libp2p/go-libp2p/core/peerstore"
 	mempstore "github.com/libp2p/go-libp2p/p2p/host/peerstore/pstoremem"
+	"golang.org/x/exp/slices"
+
+	"github.com/algorand/go-algorand/network"
```

Comment:
It looks like you have import cycles when importing Phonebook and PhonebookEntryRoles. I guess those should either move here or into a separate phonebook package